### PR TITLE
Remove /revision/{revision} endpoint

### PR DESCRIPTION
**Pull request by Pchelolo**
We've decided to remove the `/revision/{revision}` endpoint a while ago, so let's finally get rid of it.

WIP-ish we can't really release it like this, because change-prop depends on this endpoint. So once the review is done I will split it into separate stages.

One additional thing that should be done is to test it together with #718 

cc @wikimedia/services

### CODE REVIEW

**Comment by Pchelolo** [comment]
Run all the tests with the restriction_table access checks and all the tests pass!

**Comment by gwicke** [comment]
:+1: 

Any objections to merging this right away?

File: sys/page_revisions.js [hunk 1]
```diff
@@ -83,15 +83,7 @@ class PRS {
...
-            ],
-            secondaryIndexes: {
-                by_rev: [
-                    { attribute: 'rev', type: 'hash' },
-                    { attribute: 'tid', type: 'range', order: 'desc' },
-                    { attribute: 'title', type: 'range', order: 'asc' },
-                    { attribute: 'restrictions', type: 'proj' }
-                ]
-            }
+            ]
```
Comment (d00rman) [inline sys/page_revisions.js]: Should we also bump the version since it has changed?

File: sys/page_revisions.js [hunk 2]
```diff
@@ -269,110 +261,117 @@ class PRS {
...
-        const getLatestTitleRevision = () => {
-            return hyper.get({
+        const titleRevisionRequest = () => {
+            const revReqObject = {
                 uri: this.tableURI(rp.domain),
                 body: {
                     table: tableName,
                     attributes: {
-                        title: rp.title
+                        title: rp.title,
```
Comment (d00rman) [inline sys/page_revisions.js]: No comma needed at the end

**Comment by d00rman** [comment]
LGTM overall, a couple of nits in-lined. Also, needs a rebase.

> WIP-ish we can't really release it like this, because change-prop depends on this endpoint.

We are good on that front. As of [Gerrit 380932](https://gerrit.wikimedia.org/r/#/c/380932/), CP uses `/page/title/{title}{/revision}` for this.

**Comment by Pchelolo** [comment]
Rebased and bumped the schema version.

File: sys/page_revisions.js (hunk 1 above)
Comment (Pchelolo) [reply sys/page_revisions.js]: Did bump the version. We should not forget that the storage module doesn't actually delete the tables after deleting the index, so we will need to drop key spaces manually.

**Comment by d00rman** [comment]
LGTM :+1: Let's merge this right before deployment.

**Pull request merged**